### ISSUE TITLE
从onnx模型转换时GroupNorm算子报错

### ISSUE THREAD

**Issue by ziranpeidoujiang** [alibaba/MNN#3297]
原模型是pytorch写的，部分代码是：
class DualPathRNN(nn.Module):
    """  
    Dual-Path RNN in Separation Network.

    Args:
        d_model (int): The number of expected features in the input (input_size).
        expand (int): Expansion factor used to calculate the hidden_size of LSTM.
        bidirectional (bool): If True, becomes a bidirectional LSTM.
    """
    def __init__(self, d_model, expand, bidirectional=True):
        super(DualPathRNN, self).__init__()

        self.d_model = d_model
        self.hidden_size = d_model * expand
        self.bidirectional = bidirectional
        # Initialize LSTM layers and normalization layers
        self.lstm_layers = nn.ModuleList([self._init_lstm_layer(self.d_model, self.hidden_size) for _ in range(2)])
        self.linear_layers = nn.ModuleList([nn.Linear(self.hidden_size*2, self.d_model) for _ in range(2)])
        self.norm_layers = nn.ModuleList([nn.GroupNorm(1, d_model) for _ in range(2)])

    def _init_lstm_layer(self, d_model, hidden_size):
        return LSTM(d_model, hidden_size, num_layers=1, bidirectional=self.bidirectional, batch_first=True)

    def forward(self, x):
        B, C, F, T = x.shape
        
        # Process dual-path rnn
        original_x = x
        # Frequency-path
        x = self.norm_layers[0](x)
        x = x.transpose(1, 3).contiguous().view(B * T, F, C)  
        x, _ = self.lstm_layers[0](x)
        x = self.linear_layers[0](x)
        x = x.view(B, T, F, C).transpose(1, 3)
        x = x + original_x

        original_x = x
        # Time-path
        x = self.norm_layers[1](x)
        x = x.transpose(1, 2).contiguous().view(B * F, C, T).transpose(1, 2)
        x, _ = self.lstm_layers[1](x)
        x = self.linear_layers[1](x)
        x = x.transpose(1, 2).contiguous().view(B, F, C, T).transpose(1, 2)
        x = x + original_x

        return x
    

class SeparationNet(nn.Module):
    """
    Implements a simplified Sparse Down-sample block in an encoder architecture.
    
    Args:
    - channels (int): Number input channels.
    - expand (int): Expansion factor used to calculate the hidden_size of LSTM.
    - num_layers (int): Number of dual-path layers.
    """
    def __init__(self, channels, expand=1, num_layers=6):
        super(SeparationNet, self).__init__()
        
        self.num_layers = num_layers

        self.dp_modules = nn.ModuleList([
                DualPathRNN(channels * (2 if i % 2 == 1 else 1), expand) for i in range(num_layers)
        ])
        
        self.feature_conversion = nn.ModuleList([
            FeatureConversion(channels * 2 , inverse = False if i % 2 == 0 else True) for i in range(num_layers)
        ])
    def forward(self, x):
        for i in range(self.num_layers):
           x = self.dp_modules[i](x)
           x = self.feature_conversion[i](x)
        return x
 
 
 
 
 
导出为onnx成功后再使用mnn转换，报错信息如下：
[15:57:06]===> compute shape: /separation_net/dp_modules.1/norm_layers.0/Reshape_1_output_0, [Reshape]
[15:57:06]        Inputs:
[15:57:06]        ptr=0x750be00, format=NCHW, datatype=2; 1, 1, 3472896, 
[15:57:06]        ptr=0x7569550, format=NCHW, datatype=0; 4, 
[15:57:06]        Outputs:
[15:57:06]        ptr=:0x7569f40, format=NCHW, datatype=2;        1, 128, 57, 476, 
[15:57:06]Broad cast error, dim1 = 256, dim2 = 128
 
 
 
 
 
加了打印看了下GroupNorm的参数和通道数也对应得上，mnn怎么就转换失败呢？
SCNet DualPathRNN self.norm_layers[0]:  GroupNorm(1, 128, eps=1e-05, affine=True)
SCNet DualPathRNN x 1:  torch.Size([1, 128, 57, 876])
SCNet DualPathRNN self.norm_layers[1]:  GroupNorm(1, 128, eps=1e-05, affine=True)
SCNet DualPathRNN x 2:  torch.Size([1, 128, 57, 876])
SCNet DualPathRNN self.norm_layers[0]:  GroupNorm(1, 256, eps=1e-05, affine=True)
SCNet DualPathRNN x 1:  torch.Size([1, 256, 57, 439])
SCNet DualPathRNN self.norm_layers[1]:  GroupNorm(1, 256, eps=1e-05, affine=True)
SCNet DualPathRNN x 2:  torch.Size([1, 256, 57, 439])
........

**Comment by jxt1234** [comment]
先确保 mnn 是最新版本，然后用 testMNNFromOnnx.py 测试

**Comment by ziranpeidoujiang** [comment]
是最新的，因为有自定义算子存在，我用testMNNFromOnnx.py报错了，这块我明明已经按照自定义算子的指导实现过了。
我先把onnx文件和修改过后的MNN代码发出来吧：
https://1drv.ms/f/c/ea17dde0313a8ce6/EpI4nEhK65ROiFw7ICUPHfwBwuco5yy0umSZc6ESC-usPg?e=IgI8q0

**Comment by ziranpeidoujiang** [comment]
编译环境是Ubuntu 18 4.4.0

**Comment by jxt1234** [comment]
可以换成 google 网盘，或者发我邮箱么？ 这个链接我访问不了

**Comment by ziranpeidoujiang** [comment]
> 可以换成 google 网盘，或者发我邮箱么？ 这个链接我访问不了

好的，刚刚已经发送到您邮箱了

**Comment by jxt1234** [comment]
已经收到

**Comment by jxt1234** [comment]
感觉是新加的 FeatureConversion shape 算错了吧，算出来的 channel 应该是 256 而不是 128 ?

**Comment by ziranpeidoujiang** [comment]
> 感觉是新加的 FeatureConversion shape 算错了吧，算出来的 channel 应该是 256 而不是 128 ?

应该是您说的这个原因，这是要改ShapeMyCustomOp.cpp吗？
但我故意改错别的dim报错提示也不变，添加了MNN_PRINT再去跑也没看到任何打印，我这是没注册成功吗？

**Comment by ziranpeidoujiang** [comment]
找到原因了，少跑了python3 tools/script/register.py .
确实是没注册进去，导致输出shape跟输入一致了，现在可以了
